### PR TITLE
 WFLY-12072 Allow setting create-on-start / drop-on-exit on Infinispan JDBC tables

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-12072

### CODE REVIEW

**Comment by pferraro** [comment]
Retest this please.

**Review by darranl — COMMENTED** [review]
For changes like this it would be preferable for the version bump to be in a separate commit.

Looking at the changes in GitHub it is not actually possible to see things like the changes made to the schema as they are made in a new file - also in the future if anyone it looking at the history they will not be able to tell the difference between the new schema and the changed made.

**Comment by pferraro** [comment]
@darranl Makes sense. I'll move that to a separate commit.

**Comment by pferraro** [comment]
@darranl Done.

**Comment by pferraro** [comment]
Needs model version bump, since this didn't make it into WF17.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TableResourceDefinition.java [hunk 1]
```diff
@@ -59,6 +61,8 @@ static PathElement pathElement(String value) {
 
     enum Attribute implements org.jboss.as.clustering.controller.Attribute {
         FETCH_SIZE("fetch-size", ModelType.INT, new ModelNode(100)),
+        CREATE_ON_START("create-on-start", ModelType.BOOLEAN, new ModelNode(true)),
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TableResourceDefinition.java]: ModelNode.TRUE

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TableResourceDefinition.java [hunk 2]
```diff
@@ -59,6 +61,8 @@ static PathElement pathElement(String value) {
...
+        DROP_ON_STOP("drop-on-stop", ModelType.BOOLEAN, new ModelNode(false)),
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TableResourceDefinition.java]: ModelNode.FALSE

I bulk corrected hundreds of these so I'm being picky. ;)

File: clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 3]
```diff
@@ -299,6 +299,8 @@ infinispan.store.jdbc.binary-keyed-table.deprecated=Deprecated. Use table=binary
 infinispan.store.jdbc.binary-keyed-table.table.prefix=The prefix for the database table name.
 infinispan.store.jdbc.binary-keyed-table.table.batch-size=For DB inserts, the batch size determines how many inserts are batched together.
 infinispan.store.jdbc.binary-keyed-table.table.fetch-size=For DB queries, the fetch size will be used to set the fetch size on ResultSets.
+infinispan.store.jdbc.binary-keyed-table.table.create-on-start=Indicates whether the store should create this database table the cache starts.
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: s/table the/table when the/g

File: clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 4]
```diff
@@ -316,6 +318,8 @@ infinispan.store.jdbc.string-keyed-table.deprecated=Deprecated. Use table=string
 infinispan.store.jdbc.string-keyed-table.table.prefix=The prefix for the database table name.
 infinispan.store.jdbc.string-keyed-table.table.batch-size=For DB inserts, the batch size determines how many inserts are batched together.
 infinispan.store.jdbc.string-keyed-table.table.fetch-size=For DB queries, the fetch size will be used to set the fetch size on ResultSets.
+infinispan.store.jdbc.string-keyed-table.table.create-on-start=Indicates whether the store should create this database table the cache starts.
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: s/table the/table when the/g

File: clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 5]
```diff
@@ -343,6 +347,8 @@ infinispan.table.prefix=The prefix for the database table name.
 infinispan.table.batch-size=For DB inserts, the batch size determines how many inserts are batched together.
 infinispan.table.batch-size.deprecated=Deprecated. Use max-batch-size instead.
 infinispan.table.fetch-size=For DB queries, the fetch size will be used to set the fetch size on ResultSets.
+infinispan.table.create-on-start=Indicates whether the store should create this database table the cache starts.
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: s/table the/table when the/g

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_9_0.xsd [hunk 6]
```diff
@@ -747,6 +747,16 @@
                 <xs:documentation>The fetch size used when querying from this table.  Used to avoid heap memory exhaustion when query is large.</xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="create-on-start" type="xs:boolean" default="true">
+            <xs:annotation>
+                <xs:documentation>Indicates whether the store should create this database table the cache starts.</xs:documentation>
```
Comment (bstansberry) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_9_0.xsd]: s/table the/table when the/g

**Review by bstansberry — CHANGES_REQUESTED** [review]
@pferraro I'm requesting some minor things, mostly because a new CI run would be good.

If you update those and it passes CI please feel free to merge.

**Pull request merged**